include/eventfilters.h

```diff
@@ -0,0 +1,59 @@
+#ifndef EVENTFILTERS_H
+#define EVENTFILTERS_H
+
+#include <QEvent>
+#include <QPlainTextEdit>
+#include <QLineEdit>
+
+class AOTextEditFilter : public QObject
+{
+    Q_OBJECT
+
+protected:
+    // QPlainTextEdit's viewport is its own object and it receives mouse events instead of the main object
+    // so we are actually intercepting events meant for the viewport and working w/ its parent (the object we want) instead
+    //
+    // this took too long to figure out. QPlainTextEdit more like QPainTextEdit
+    bool eventFilter(QObject *obj, QEvent *event) override {
+        if (event->type() == QEvent::MouseButtonDblClick && obj->parent() != nullptr) {
+            QPlainTextEdit *textEdit = qobject_cast<QPlainTextEdit *>(obj->parent());
+            if (textEdit != nullptr)
+                textEdit->setReadOnly(false);
```

Comment:
True. My focus with this was on reimplementing existing behavior, but it would probably be a good idea to improve on it too. I like the padlock idea, think I'll use that.